src/coreclr/gc/gc.cpp

```diff
@@ -25152,37 +25165,90 @@ float median_of_3 (float a, float b, float c)
     return b;
 }
 
-size_t gc_heap::get_num_completed_gcs ()
+float log_with_base (float x, float base)
 {
-    size_t num_completed_gcs = settings.gc_index;
-#ifdef BACKGROUND_GC
-    if (g_heaps[0]->is_bgc_in_progress ())
+    assert (x > base);
+
+    return (float)(log(x) / log(base));
+}
+
+float mean (float* arr, int size)
+{
+    float sum = 0.0;
+
+    for (int i = 0; i < size; i++)
     {
-        num_completed_gcs--;
-        dprintf (6666, ("BGC in prog, completed GCs -> %Id", num_completed_gcs));
+        sum += arr[i];
     }
-#endif //BACKGROUND_GC
+    return (sum / size);
+}
+
+// Change it to a desired number if you want to print.
+int max_times_to_print_tcp = 0;
+
+// Return the slope, and the average values in the avg arg.
```

Comment:
Is there a name for the slope that is being calculated here?  I see that it's a weighted sum based on distance from the middle, but I'm not familiar with that.  For example, I don't think this is the slope of a typical regression line?  (which is fine, though I guess I'm a bit curious about the mathematical properties of this)